hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -323,4 +322,31 @@ private void registerConfiguredNamedSerializers(CompactSerializationConfig compa
             typeNameToRegistrationMap.put(typeName, registration);
         }
     }
+
+    public Schema extractSchema(BufferObjectDataInput objectDataInput) throws IOException {
+        return getOrReadSchema(objectDataInput, false);
+    }
+
+    public Schema extractSchema(Object o) {
+        CompactSerializableRegistration registration = getOrCreateRegistration(o);
+        Class<?> aClass = o.getClass();
+
+        Schema schema = classToSchemaMap.get(aClass);
+        if (schema == null) {
+            schema = buildSchema(registration, o);
+            return schema;
+        }
+        return schema;
+    }
+
+    private static Schema buildSchema(CompactSerializableRegistration registration, Object o) {
+        SchemaWriter writer = new SchemaWriter(registration.getTypeName());
+        try {
+            registration.getSerializer().write(writer, o);
+        } catch (IOException e) {
+            //Schema writer does not throw IOException
+            EmptyStatement.ignore(e);
+        }
```

Comment:
Minor: I feel like it would be better to put the `IOException` to the method signature. The places where we use this either also throws `IOException` or handle that, so no code change would be required for them, and we would get rid of this try-catch block. (Of course, we need to change the signature of `Schema extractSchema(Object o)` but I think it is OK)

Reply:
While doing this I have realized that CompactSerializer is throwing IOException unncessarily. CompactReader and CompactWriter does not. I plan to send a second pr that will remote IOException from `CompactSerializer` API. 
So I will remove this a couple of lines back on this pr back.